libkbfs/interfaces.go

```diff
@@ -1285,6 +1295,19 @@ type DiskMDCache interface {
 	Shutdown(ctx context.Context)
 }
 
+// DiskBlockMetadataStore defines a type that stores xattrs.
+type DiskBlockMetadataStore interface {
+	// GetXattr looks for and returns the Xattr value of xattrType for blockID
+	// if it's found, and ldberrors.ErrNotFound if it's not found.
+	GetXattr(ctx context.Context,
+		blockID kbfsblock.ID, xattrType XattrType) ([]byte, error)
+	// SetXattr sets xattrType Xattr to xattrValue for blockID.
+	SetXattr(ctx context.Context,
+		blockID kbfsblock.ID, xattrType XattrType, xattrValue []byte) error
```

Comment:
Why wouldn't you return a `blockMetadata` struct instead of an `interface{}` or `[]byte`? I mean ideally you wouldn't even need the `Xattr` methods at all, and you could pull them out of the `blockMetadata` struct. But I understand that you want the `Noop` shim, which is why I went with this recommendation.

Reply:
Ah that's right; that'd work. Not sure what I was thinking.

There's one more thing though. In the "set" path, it's actually an update. With `SetXattr`, caller just need to pass in the xattr value. Otherwise, caller need to always do a `GetMetadata` first, amend the value, then call `SetMetadata`. This is what happens under-the-hood anyways, but I'd love the encapsulation that `SetXattr` provides.

Sounds like the suggested approach from your original comment works the best then. Will update; thanks!

Reply:
Hmm actually in the "setXattr" path, getMetadata and setMetadata need to happen in the same locking session. So the presumed `SetMetadata` would probably have an interface that looks like the `setMetadata` right now. Does it sound good to you? @jzila

Reply:
Yep, I'm fine with that.